SmartDeviceLink/private/SDLPresentAlertOperation.m

```diff
@@ -0,0 +1,463 @@
+
+@end
+
+@interface SDLPresentAlertOperation()
+
+@property (strong, nonatomic) NSUUID *operationId;
+@property (weak, nonatomic) id<SDLConnectionManagerType> connectionManager;
+@property (weak, nonatomic) SDLFileManager *fileManager;
+@property (weak, nonatomic) SDLSystemCapabilityManager *systemCapabilityManager;
+@property (strong, nonatomic, readwrite) SDLAlertView *alertView;
+@property (assign, nonatomic) UInt16 cancelId;
+@property (copy, nonatomic, nullable) NSError *internalError;
+
+@end
+
+@implementation SDLPresentAlertOperation
+
+#pragma mark - Lifecycle
+
+- (instancetype)initWithConnectionManager:(id<SDLConnectionManagerType>)connectionManager fileManager:(SDLFileManager *)fileManager systemCapabilityManager:(SDLSystemCapabilityManager *)systemCapabilityManager currentWindowCapability:(nullable SDLWindowCapability *)currentWindowCapability alertView:(SDLAlertView *)alertView cancelID:(UInt16)cancelID {
+
+    self = [super init];
+    if (!self) { return nil; }
+
+    _connectionManager = connectionManager;
+    _fileManager = fileManager;
+    _systemCapabilityManager = systemCapabilityManager;
+
+    __weak typeof(self) weakSelf = self;
+    alertView.canceledHandler = ^{
+        [weakSelf sdl_cancelInteraction];
+    };
+    _alertView = [alertView copy];
+
+    _cancelId = cancelID;
+    _operationId = [NSUUID UUID];
+    _currentWindowCapability = currentWindowCapability;
+
+    return self;
+}
+
+- (void)start {
+    [super start];
+    if (self.isCancelled) { return; }
+
+    NSError *alertViewValidatedError = [self sdl_isValidAlertViewData:self.alertView];
+    if (alertViewValidatedError) {
+        [self finishOperation];
+        self.internalError = alertViewValidatedError;
+        return;
+    }
+
+    dispatch_group_t uploadFilesTask = dispatch_group_create();
+    dispatch_group_enter(uploadFilesTask);
+
+    dispatch_group_enter(uploadFilesTask);
+    [self sdl_uploadImagesWithCompletionHandler:^{
+        dispatch_group_leave(uploadFilesTask);
+    }];
+
+    dispatch_group_enter(uploadFilesTask);
+    [self sdl_uploadAudioFilesWithCompletionHandler:^{
+        dispatch_group_leave(uploadFilesTask);
+    }];
+
+    dispatch_group_leave(uploadFilesTask);
+     
+    // This will always run after all `leave`s
+    __weak typeof(self) weakSelf = self;
+    dispatch_group_notify(uploadFilesTask, [SDLGlobals sharedGlobals].sdlConcurrentQueue, ^{
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        [strongSelf sdl_presentAlert];
+    });
+}
+
+/// Checks the `AlertView` data to make sure it conforms to the RPC Spec, which says that at least either `alertText1`, `alertText2` or `TTSChunks` need to be provided.
+/// @return The error if the alert view does not have valid data; nil if the alert view data is valid
+- (nullable NSError *)sdl_isValidAlertViewData:(SDLAlertView *)alertView {
+    BOOL isValidData = NO;
+    if (alertView.text.length > 0) {
+        isValidData = YES;
```

Comment:
I don't think so...since `isValidData` is initially set to `NO` and is only set to `YES` if one of the conditions is met (i.e. `text.length > 0`, `secondaryText.length > 0` or at least one TTS chunk is set. 

I could return `nil` early, but I thought it best to consolidate that logic in the `if/else` statement.